gemd/units/tests/test_parser.py

```diff
@@ -39,3 +40,19 @@ def test_parse_unexpected():
 def test_parse_none():
     """Test that None parses as None."""
     assert parse_units(None) is None
+
+
+def test_file_change():
+    """Test that swapping units files works."""
+    assert convert_units(1, 'm', 'cm') == 100
+    with pytest.raises(UndefinedUnitError):
+        assert convert_units(1, 'usd', 'usd') == 1
+    change_definitions_file(
+        filename=pkg_resources.resource_filename("gemd.units", "tests/test_units.txt")
+    )
+    with pytest.raises(UndefinedUnitError):
+        assert convert_units(1, 'm', 'cm') == 100
+    assert convert_units(1, 'usd', 'usd') == 1
+    change_definitions_file(
```

Comment:
If this change back is to make sure this test state doesn't pollute other tests, then it would be best to wrap the rest of the test in a `try... except ... finally` block and stick the convert back into the `finally`.  Even better would be to use a context manager, which provides the same behavior but without relying on `try... except`

Reply:
Good catch